drivers/overlay/overlay.go

```diff
@@ -995,9 +995,13 @@ func (d *Driver) get(id string, disableShifting bool, options graphdriver.MountO
 		return "", fmt.Errorf("error creating overlay mount to %s: %v", mountTarget, err)
 	}
 
-	// chown "workdir/work" to the remapped root UID/GID. Overlay fs inside a
-	if err := os.Chown(path.Join(workDir, "work"), rootUID, rootGID); err != nil {
-		return "", err
+	workPath := path.Join(workDir, "work")
+	stat, err := system.Stat(workPath)
+	if err != nil || stat.UID() != uint32(rootUID) || stat.GID() != uint32(rootGID) {
+		// chown "workdir/work" to the remapped root UID/GID. Overlay fs inside a
+		if err := os.Chown(workPath, rootUID, rootGID); err != nil {
+			return "", err
+		}
```

Comment:
This looks like a good place to replace `os.Chown()` with a call to the updated `idtools.SafeChown()`.

Reply:
I think it would be faster if you try the `SafeChown` in any case then check whether it was needed only on error:
```
if err := SafeChown(...); err != nil {
    stat, err := system.Stat(workPath)
    ... raise the error only if uid!=stat.uid || gid!=stat.gid, otherwise ignore it
}
```

Reply:
Why, I would figure it would be much more often the case where the file already exists with root, root and the chown takes more time.

Reply:
@nalind Changes to use SafeChown for overlay.

Reply:
> Why, I would figure it would be much more often the case where the file already exists with root, root and the chown takes more time.

ok then in that case it is fine.  I assumed we were chown'ing most of the time